### PR TITLE
Client

### PR DESCRIPTION
**Pull request by saladuit**
- Added 404.html file
- Added extra client states to create a distinction between different out modes
- Added get, post and delete routines in the file manager
- Created a Error loading mechanism in the file manager; loads file when defined, otherwise makes a default html status
- Created a poll exception class
- Created a StatusCode enum class for all of the status codes
- Added the client handling logic in handleActiveConnection method, which tracks and returns the client state to the HTTPServer class.
- And a lot more

### CODE REVIEW

File: src/HTTPResponse.cpp [hunk 1]
```diff
@@ -1,27 +1,54 @@
+#include "HTTPRequest.hpp"
```
Comment (Tentanus) [inline src/HTTPResponse.cpp:1]: change to < >

File: src/HTTPServer.cpp [hunk 2]
```diff
@@ -28,11 +28,10 @@ int HTTPServer::run()
 		logger.log(INFO, "Server started");
 		while (true)
```
Comment (Tentanus) [inline src/HTTPServer.cpp:29]: Does this belong in the Try/Catch block.

File: src/Socket.cpp [hunk 3]
```diff
@@ -13,9 +14,9 @@ Socket::Socket(const int fd)
 {
```
Comment (Tentanus) [inline src/Socket.cpp:14]: change name fd to server_fd and it shouldn't be const

Comment (Tentanus) [reply src/Socket.cpp:14]: also the inet_ntop shouldn't be done in the constructor

**Review by Tentanus — CHANGES_REQUESTED** [review]
Great job, Massive Pull request. it's a big boy

File: src/FileManager.cpp [hunk 4]
```diff
@@ -1,134 +1,146 @@
...
-#include <fstream>
-#include <iostream>
-#include <string>
 
-namespace fs = std::filesystem;
+#include <algorithm>
+#include <filesystem>
 
-FileManager::FileManager() : _status_code(0)
+FileManager::FileManager()
```
[the remark this answers is not in the document]
Comment (saladuit) [reply src/FileManager.cpp:8]: Added However unsure, whether the file gets initialised correctly

File: src/HTTPRequest.cpp [hunk 5]
```diff
@@ -1,14 +1,14 @@
...
+
+#include <string>
 
 #include <unistd.h>
 
 HTTPRequest::HTTPRequest()
-	: _bytes_read(0), _methodType(HTTPMethod::UNKNOWN), _http_request(),
-	  _headers()
+	: _bytes_read(0), _content_length(0), _methodType(HTTPMethod::UNKNOWN),
+	  _http_request(), _body(), _headers()
```
[the remark this answers is not in the document]
Comment (saladuit) [reply src/HTTPRequest.cpp:11]: added the other variables as well

File: src/HTTPServer.cpp (hunk 2 above)
Comment (saladuit) [reply src/HTTPServer.cpp:29]: made a seperate try catch block which doesn't stop the server

File: src/Socket.cpp (hunk 3 above)
Comment (saladuit) [reply src/Socket.cpp:14]: made a seperate funciton that sets up the client socket, which is called in the constructor of the client class that it only happens after the whole socket is already created.

Comment (Tentanus) [reply src/Socket.cpp:14]: great work

**Review by Tentanus — APPROVED** [review]
Sicko good job!

**Pull request merged**